src/components/AddRemoteDialogue.tsx

```diff
@@ -0,0 +1,191 @@
+import { Git } from '../tokens';
+import { GitExtension } from '../model';
+import {
+  remoteDialogClass,
+  remoteDialogInputClass,
+  existingRemoteListClass,
+  existingRemoteItemClass
+} from '../style/AddRemoteDialog';
+import { TranslationBundle } from '@jupyterlab/translation';
+
+import { classes } from 'typestyle';
+import {
+  actionsWrapperClass,
+  buttonClass,
+  cancelButtonClass,
+  closeButtonClass,
+  contentWrapperClass,
+  createButtonClass,
+  titleClass,
+  titleWrapperClass
+} from '../style/NewBranchDialog';
+
+import { deletionsMadeIcon } from '../style/icons';
+
+export interface IAddRemoteDialogueProps {
+  /**
+   * The application language translator.
+   */
+  trans: TranslationBundle;
+  /**
+   * Warning content.
+   */
+  warningContent?: string;
+  /**
+   * Git extension model
+   */
+  model: GitExtension;
+  /**
+   * Callback to handle the closing of dialogue
+   */
+  onClose: (remote?: Git.IGitRemote) => void;
+}
+
+export interface IAddRemoteDialogueState {
+  /**
+   * New remote name and url pair
+   */
+  newRemote: Git.IGitRemote;
+  /**
+   * List of known remotes
+   */
+  existingRemotes: Git.IGitRemote[];
+}
+
+export class AddRemoteDialogue extends React.Component<
+  IAddRemoteDialogueProps,
+  IAddRemoteDialogueState
+> {
+  constructor(props: IAddRemoteDialogueProps) {
+    super(props);
+    this.state = {
+      newRemote: {
+        name: '',
+        url: ''
+      },
+      existingRemotes: []
+    };
+  }
+
+  async componentDidMount(): Promise<void> {
+    const remotes = await this.props.model.getRemotes();
+    this.setState({ existingRemotes: remotes });
+  }
+
+  render(): JSX.Element {
+    return (
+      <Dialog
+        classes={{
+          paper: remoteDialogClass
+        }}
+        open={true}
+        onClose={this.props.onClose}
+      >
+        <div className={titleWrapperClass}>
+          <p className={titleClass}>{this.props.trans.__('Manage Remotes')}</p>
+          <button className={closeButtonClass}>
+            <ClearIcon
+              titleAccess={this.props.trans.__('Close this dialog')}
+              fontSize="small"
+              onClick={() => {
+                this.props.onClose();
+              }}
+            />
+          </button>
+        </div>
+        <div className={contentWrapperClass}>
+          <label className={remoteDialogInputClass}>
+            <span>
+              {this.props.trans.__(
+                'Enter a new remote repository name and url'
+              )}
+            </span>
+            <input
+              type="text"
+              placeholder="name"
+              onChange={event =>
+                this.setState({
+                  newRemote: {
+                    ...this.state.newRemote,
+                    name: event.target.value
+                  }
+                })
+              }
+            />
+            <input
+              type="text"
+              placeholder="Remote Git repository URL"
```

Comment:
That should be translated too.

```suggestion
              placeholder={this.props.trans.__('Remote Git repository URL')}
```